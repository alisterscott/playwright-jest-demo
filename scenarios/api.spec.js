/* eslint-env jest */
const supertest = require('supertest')
const jestTimeoutMS = require('config').get('jestTimeoutMS')

test('can GET a REST API and check response using approval style', async () => {
  const request = supertest('https://my-json-server.typicode.com/webdriverjsdemo/webdriverjsdemo.github.io')
  const response = await request.get('/posts')
  expect(response.status).toBe(200)
  expect(response.body).toMatchSnapshot()
}, jestTimeoutMS)

test('can GET a REST API and check response using assertion style', async () => {
  const request = supertest('https://my-json-server.typicode.com/webdriverjsdemo/webdriverjsdemo.github.io')
  const response = await request.get('/posts')
  expect(response.status).toBe(200)
  expect(response.body.length).toBe(3)
  expect(response.body[0].id).toBe(1)
  expect(response.body[0].title).toBe('Post 1')
  expect(response.body[1].id).toBe(2)
  expect(response.body[1].title).toBe('Post 2')
  expect(response.body[2].id).toBe(3)
  expect(response.body[2].title).toBe('Post 3')
}, jestTimeoutMS)

test('can POST a REST API and check response using approval style', async () => {
  const request = supertest('https://my-json-server.typicode.com/webdriverjsdemo/webdriverjsdemo.github.io')
  const response = await request.post('/posts').send({ title: 'Post 4' })
  expect(response.status).toBe(201)
  expect(response.body).toMatchSnapshot()
}, jestTimeoutMS)

test('can POST a REST API and check response using assertion style', async () => {
  const request = supertest('https://my-json-server.typicode.com/webdriverjsdemo/webdriverjsdemo.github.io')
  const response = await request.post('/posts').send({ title: 'Post 4' })
  expect(response.status).toBe(201)
  expect(response.body.id).toBe(4)
  expect(response.body.title).toBe('Post 4')
}, jestTimeoutMS)
