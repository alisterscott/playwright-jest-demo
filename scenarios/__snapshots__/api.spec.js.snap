// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can GET a REST API and check response using approval style 1`] = `
Array [
  Object {
    "id": 1,
    "title": "Post 1",
  },
  Object {
    "id": 2,
    "title": "Post 2",
  },
  Object {
    "id": 3,
    "title": "Post 3",
  },
]
`;

exports[`can POST a REST API and check response using approval style 1`] = `
Object {
  "id": 4,
  "title": "Post 4",
}
`;
